analyzers/src/SonarAnalyzer.Common/Rules/InsteadOfAnyBase.cs

```diff
@@ -0,0 +1,124 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2023 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+namespace SonarAnalyzer.Rules;
+
+public abstract class InsteadOfAnyBase<TSyntaxKind, TInvocationExpression> : SonarDiagnosticAnalyzer
+    where TSyntaxKind : struct
+    where TInvocationExpression : SyntaxNode
+{
+    private const string ExistsDiagnosticId = "S6605"; // Collection-specific "Exists" method should be used instead of the "Any" extension.
```

Comment:
Comment is slightly too long, you can just write something like:

```csharp
private const string ExistsDiagnosticId = "S6605"; // "Exists"
```

Reply:
This was just a reference for me during the implementation, I removed those entirely 👍